packages/fuels-types/src/coin.rs

```diff
@@ -37,12 +29,12 @@ impl From<ClientCoin> for Coin {
     fn from(coin: ClientCoin) -> Self {
         Self {
             amount: coin.amount.0,
-            block_created: coin.block_created.0,
+            block_created: coin.block_created.0.into(),
             asset_id: coin.asset_id.0 .0,
             utxo_id: coin.utxo_id.0 .0,
-            maturity: coin.maturity.0,
+            maturity: coin.maturity.0.into(),
             owner: coin.owner.0 .0.into(),
-            status: coin.status.into(),
+            status: CoinStatus::Unspent,
```

Comment:
The `fuel-core` itself doesn't plan to return spent coins in the future. But the indexer, for example, may return them. So maybe it is better to keep this field for future integration with the indexer

Reply:
Our thinking was that we won't ever receive spent coins from any of the endpoints provided by `fuel-core`. We also mention in the documentation, almost every time, that the returned coins would always be *unspent* ones.

When the user gets the coins he can still check to see whether they are spent or not. Even though by design they are not.

Kinda gives confusing legacy vibes to the design, to have fields without meaning for all of the users of the provider API. Coin/message status isn't relevant anywhere else in the SDK either.

A similar situation arose recently with the reexport of the fuel transaction, for which the direct users of the SDK have no need but the indexer does. So most of our users would use the enriched transaction while the indexer project uses the fuel-tx one. 

Maybe we should have the indexer write for their users a wrapper around the coin containing the status? 

Or maybe we should do it but take care to separate indexer functionality in the SDK whenever it starts to degrade the design for the remaining majority of users?

Reply:
It depends on how we plan to use SDK and indexer. I thought that the user will use SDK to query `fuel-indexer`, which is why types in SDK should be oriented to fetch data from the indexer.

But if we plan that indexer use SDK to request data from `fuel-core`, then yes, we need to use types oriented to the `fuel-core` functionality. But in those case how we can fetch data from the indexer?

The SDK can try to support interactions with `fuel-core` and with `fuel-indexer`, but it means we need different types for each layer.

Reply:
I'm not sure how far along are the plans to shift the API over to the indexer. If/when that happens I'm not sure how much of the current functionality will the fuel core retain. i.e. will our users need to get some info via the fuel core API and the rest through the future indexer API? Will those overlap with mandatory support for both from the SDK?

In any case, it seems too soon to know for sure. We can leave the field and see if the indexer will be able to use it as-is. Revisit this later once we start polling the indexer for data.

Reply:
In the future, we will have one endpoint that returns all information from the indexer and `fuel-core`. Maybe it will be part of the indexer itself or a dedicated application that merges all endpoints into one.

@ra0x3 Do you need a SDK to request data from the `fuel-core`, our `fuel-core-client` is enough for you?